src/btree/bt_split.c

```diff
@@ -1180,16 +1165,13 @@ __split_internal(WT_SESSION_IMPL *session, WT_PAGE *parent, WT_PAGE *page)
 	WT_ASSERT(session, WT_INTL_INDEX_GET_SAFE(page) == pindex);
 	WT_INTL_INDEX_SET(page, replace_index);
 
+	/* The split is live. */
+	WT_WRITE_BARRIER();
```

Comment:
There's already a write barrier in `WT_INTL_INDEX_SET` before the pointer is published, which I think it is in the right order.  In other words, it guarantees that everything in the allocated index is set *before* the pointer becomes visible to other threads.

What does a barrier after the pointer is visible achieve?